src/container/app.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component} from 'react'
+import Map from '../component/map'
+import {TextField, List, ListItem} from '@material-ui/core'
+import { withStyles } from '@material-ui/core/styles'
+import PlacesAutocomplete, {
+  geocodeByAddress,
+  getLatLng, loading
+} from 'react-places-autocomplete'
+import {connect} from 'react-redux'
+import {setCenter} from '../store/action/map'
+import CourseList from '../component/courseList'
+
+const styles = theme => ({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  },
+  textField: {
+    marginLeft: theme.spacing.unit,
+    marginRight: theme.spacing.unit
+  },
+  add: {
+    float: 'right',
+    height: 80
+  },
+  title: {
+    marginTop: '3.5rem',
+    marginBottom: '2.5rem'
+  }
+})
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      currentPosition: {
+        lat: 0,
+        lng: 0
+      },
+      address: '',
+      courses: null
+    }
+    this.handleChange = this.handleChange.bind(this)
+    this.hanleSelect = this.handleSelect.bind(this)
+    this.handleCourseInfo = this.handleCourseInfo.bind(this)
+  }
+
+  componentDidMount() {
+    navigator.geolocation.getCurrentPosition(position => {
+      this.setState({
```

Comment:
`setState` takes a callback so you don't have to get the location twice. You can wait until state is updated with the location the first time.
https://reactjs.org/docs/react-component.html#setstate